### ISSUE TITLE
exported ccmt.json from main site can't be imported to children site in a multi-site network

### ISSUE THREAD

**Issue by GoogleCodeExporter** [viktorthang/wordpress-custom-content-type-manager#211]
```
What steps will reproduce the problem? (Be specific)
1.exported ccmt.json from main site
2. import it in another children site
3. failed

What is the expected output? What do you see instead?
should import successfullly....i saw warning message on screen instead.


Please provide any additional information below (e.g. Browser, version,
OS):

*SYSTEM INFO* 
------------------------ 
Plugin Version: 0.9.4.2-pl
WordPress Version: 3.2.1
PHP Version: 5.3.5
MySQL Version: 5.5.8-log
Server OS: WINNT
------------------------ 
Other Active plugins: 

Adminimize
Admin Management Xtended
Admin Menu Editor
Advanced Access Manager
Theme My Login
WPtouch Pro

below are detail warning message,wish helpful to you:

"
Please correct the following errors:

The uploaded file is not in the correct format."



"
Warning: Invalid argument supplied for foreach() in 
C:\wamp\www\wp-content\plugins\custom-content-type-manager\views\import.php on 
line 45
Call Stack
#   Time    Memory  Function    Location
1   0.0011  462312  {main}( )   ..\admin.php:0
2   3.0324  36977352    do_action( )    ..\admin.php:151
3   3.0325  36978264    call_user_func_array ( )    ..\plugin.php:405
4   3.0325  36978280    CCTM::page_main_controller( )   ..\plugin.php:0
5   3.0341  37038688    include( 
'C:\wamp\www\wp-content\plugins\custom-content-type-manager\controllers\import_d
ef.php' )   ..\CCTM.php:1606
6   3.0590  37226384    CCTM::load_view( )  ..\import_def.php:79
7   3.0594  37290160    include( 
'C:\wamp\www\wp-content\plugins\custom-content-type-manager\views\import.php' 
)   ..\CCTM.php:1530
"
```

Original issue reported on code.google.com by `matado...@gmail.com` on 11 Oct 2011 at 5:25

**Comment by GoogleCodeExporter** [comment]
```
I'm still not able to duplicate this behavior.

This could have been related to issue 210, so perhaps the 0.9.4.3-dev patch 
already takes care of it (download it from http://wpcctm.com/).

```

Original comment by `fireproofsocks` on 12 Oct 2011 at 8:42
- Changed state: **InReview**
- Added labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
i downloaded the dev version....and deleted the previous 0.9.4.2 version....
then when i activate the dev version.....i got this message:

Fatal error: Class 'CCTM_checkbox' not found in 
C:\wamp\www\wp-content\plugins\custom-content-type-manager\includes\CCTM.php on 
line 634


i can't access my admin dashboard now...
```

Original comment by `matado...@gmail.com` on 12 Oct 2011 at 8:56

**Comment by GoogleCodeExporter** [comment]
```
Ack.  Ok, looking into this.
```

Original comment by `fireproofsocks` on 12 Oct 2011 at 2:43

**Comment by GoogleCodeExporter** [comment]
```
Ok, I've reworked the code re the CCTM_checkbox -- this error only appeared in 
certain cases, so I missed it before.

Committed revision 450312.

I still haven't been able to reproduce the problems with importing/exporting.  
Did you have version 0.9.4.x on both sites?  You can't import a 0.9.3.x 
definition on a 0.9.4.x site.
```

Original comment by `fireproofsocks` on 12 Oct 2011 at 4:46
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
import and export bug fixed? good news...i will try it later....
btw....can CCTM import content data from CSV format files?if not, would you 
consider this function in near future?
with this feature....people can export some data from other system and import 
it into custom post type....that's extremely userful...to a CMS, after all we 
are using wordpress as CMS rather than blog.
```

Original comment by `matado...@gmail.com` on 17 Oct 2011 at 12:52

**Comment by GoogleCodeExporter** [comment]
```
The CCTM .json files are pretty easy to import: they're small, they define the 
data structures and definitions and the data object is used only by the CCTM 
plugin.

A lot of people have asked about importing *content* via CSV files... it's not 
trivial, so I'm hoping that a paying gig comes along that requires it, because 
it's easily a couple days of work to code something like that.  The easiest 
alternative (for some cases) is doing a MySQL dump of the WP database and 
import that, but that doesn't cover cases where data has come from some other 
system.  You can't slurp the data straight (e.g. doing a MySQL "LOAD DATA LOCAL 
INFILE" statement) because WP does not store all the data in a single table: it 
uses 2 tables, so one column in the CSV might wind up being stored in multiple 
rows in the wp_postmeta table, so you must handle that programmatically, and of 
course, as soon as you're dealing with large CSVs and lengthy database queries, 
it's much more difficult to troubleshoot.  Anyway, it would be a great feature, 
I'd just need a paying gig if it's gonna make it into my tight schedule.

I released 0.9.4.4 last night, which patched up a couple things with the update 
process.
```

Original comment by `fireproofsocks` on 17 Oct 2011 at 4:19
